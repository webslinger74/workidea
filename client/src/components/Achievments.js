import React from 'react';


const Achievements = () => {
    return (
        <div>
            Achievements are cool
        </div>
      );
}
 
export default Achievements;