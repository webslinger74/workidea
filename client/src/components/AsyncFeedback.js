import Loadable from 'react-loadable';
import React from 'react';

const LoadingComponent = () => <h3>Please Wait...</h3>

const FeedbackComponentPromise = () => {
    return import('./Feedback');
}

const AsyncFeedback = Loadable({
    loader:FeedbackComponentPromise,
    loading:LoadingComponent
});

export default AsyncFeedback;
