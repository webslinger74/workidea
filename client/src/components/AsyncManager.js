import Loadable from 'react-loadable';
import React from 'react';

const LoadingComponent = () => <h3>Please Wait...</h3>

const ManagerComponentPromise = () => {
    return import('./Manager');
}

const AsyncManager = Loadable({
    loader:ManagerComponentPromise,
    loading:LoadingComponent
});

export default AsyncManager;