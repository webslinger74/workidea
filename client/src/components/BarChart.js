import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as d3 from 'd3';
import d3Tip from 'd3-tip';
 

class BarChart extends Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }


    componentDidMount(){
        this.drawChart();
    }

    componentDidUpdate(prevProps){
        if(this.props.data !== prevProps.data){
            this.drawChart();
        }

    }

    drawChart = () => {

       
        const data = this.props.data;

        const max = d3.max(data, d => d.amount);

        const margin = {
            top:20, right:20, bottom:100, left:100
        }

        const graphWidth = 600 - margin.right - margin.left;
        const graphHeight = 500 - margin.top - margin.bottom;


        const y = d3.scaleLinear()
            .domain([0,max])
            .range([graphHeight,0]);

        const x = d3.scaleBand()
            .domain(data.map(item => item.name))
            .range([0,500])
            .paddingInner(0.2)
            .paddingOuter(0.2);

       

      //  const svg = d3.select ('#barChart').append("svg").attr("width", 700).attr("height", 600);

        let u = d3.select(this.svgElememt)
        
        let gr = d3.select(this.grElement)
        
            gr.attr('width', graphWidth)
            .attr('height', graphHeight)
            .attr('transform', `translate(${margin.left}, ${margin.top})`)


        let grX = d3.select(this.grXElement);
        let grY = d3.select(this.grYElement);

        const xAxisGroup = grX
                .attr('transform', `translate(0, ${graphHeight})`);
        const yAxisGroup = grY;
                
        

        const xAxis = d3.axisBottom(x);
        const yAxis = d3.axisLeft(y);

        xAxisGroup.call(xAxis);
        yAxisGroup.call(yAxis);

        //up to here is controlled by React then kicks in the D3 parts

         
            //rects already in DOM
        const rects = gr.selectAll('rect')
            .data(data);


          /* Initialize tooltip */
       const tip = d3Tip().attr('class', 'd3-tip').html(function(d) { return d.amount; });
       /* Invoke the tip in the context of your visualization */
             rects.call(tip)   
      

            rects.exit().remove();
        
           rects.attr("fill", this.props.color)
            .attr('class', "bars")
            .attr("x", d => x(d.name))
            .attr("y", graphHeight)
            .attr("height", 0)
            .attr("width", x.bandwidth)
            .on('mouseover', tip.show)
            .on('mouseout', tip.hide)     
            .transition().duration(4000)
            .attr("height", d => graphHeight - y(d.amount))
            .attr("y", d => y(d.amount));
   /* Invoke the tip in the context of your visualization */
   rects.call(tip)   
      
   
//enter selection////////////////
        rects.enter()
        .append('rect')
                .attr("fill", this.props.color)
                .attr('class', "bars")
                .attr("x", d => x(d.name))
                .attr("y", graphHeight)
                .attr("height", 0)
                .attr("width", x.bandwidth)
                .on('mouseover', tip.show)
                .on('mouseout', tip.hide)     
                .transition().duration(4000)
                  .attr("height", d => graphHeight - y(d.amount))
                  .attr("y", d => y(d.amount));

   /* Invoke the tip in the context of your visualization */
   rects.call(tip)   
      
          
    }

      render(){
          
    return (

        

        <div>
                   
            <div className="dynamicCharts">
            <svg width={this.props.width} height={this.props.height} ref={element => this.svgElememt = element}>
                <g ref={element => this.grElement = element}>
                <g ref={element => this.grXElement = element}></g>
                <g ref={element => this.grYElement = element}></g>
                
                </g>
            </svg>
          
            </div>
         
            

       </div>
    
      
      )
    }
}



const actions = {
}

const mapStateToProps = (state) => ({

})
 
export default connect(mapStateToProps, actions)(BarChart);