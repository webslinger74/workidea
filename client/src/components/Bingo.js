import React, { Component } from 'react';
import { getBingoNumbers } from '../actions/sportsActions';
import { connect } from 'react-redux';
import Sports from './Sports';

class Bingo extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

         componentDidMount(){
            this.props.getBingoNumbers();
         }

      render(){
        const { bingo } = this.props;
        console.log(bingo, "bingo");

    return (
            <Sports>
        <div>    
        <div className="sportsItem">
        <h1>Pay Day Bingo</h1>
            <p>Come and Play Bingo Every Pay Day on Site. Â£1 a ticket with Cash prizes to be Won every Month</p>  
            <p>Local reps will be selling tickets near you in the lead up to Pay day</p>
            <p>Rules - Email BST as soon as you have all the numbers on  your purchased ticket</p>
            <p>Email with the number that you completed your card with !</p>
            <p>GOOD LUCK</p>  
            <div id="bingocard">               
                   <div className="cont">
                   { bingo && bingo.map(element => {
                    
                    return (<div id={`bingoCardInner${element.number}`} key={element._id}> {element.number}
                    </div> )
                   })
                   }
                    
                    
        </div>
        
        </div>
        </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getBingoNumbers
}

const mapStateToProps = (state) => ({
    bingo:state.sports.bingoNumbers
})
 
export default connect(mapStateToProps, actions)(Bingo);