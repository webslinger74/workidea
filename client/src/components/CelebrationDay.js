import React, { Component } from 'react';
import { connect } from 'react-redux';
import Sports from './Sports';
import ReactHtmlParser from 'react-html-parser';
import { getCelebrationDay, deleteCelebrationDay } from '../actions/sportsActions';

class CelebrationDay extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    convertStringMessageToHtml = (string) => {
        const html = string;
        return <div>{ReactHtmlParser(html)}</div>
}

showMessages = (messages) => {

    
   if(messages && messages[0]) {
   return (
         <div className="latestSportsMessage" key={messages[0]._id}>
       <div>
            <h3 className="latestMessageTitle"> {messages[0].title} </h3> </div>
            
            {messages[0].images && messages[0].images.length > 0 ?
                       (<div className="imgHolder">
                            
                               <img className="messageBoardImgFront" src={messages[0].images[0].url}></img>
                       </div>) : null
                }
              
              
            
                <div>
       {messages[0].message ? 
            this.convertStringMessageToHtml(messages[0].message):
            null}                    
            </div>
            <div>
            <h3 className="authorStamp"> {messages[0].author} - {messages[0].createdAt}  </h3>
        </div>
        {this.props.auth ?
        <div style={{paddingRight:"25px", width:"150px", fontSize:"20px", backgroundColor:"black", color:"white"}} onClick={()=> this.props.deleteCelebrationDay({id:messages[0]._id}, messages[0].images)} className="delete">Delete Celebration Day</div>
        : null } </div>
          
     )
  
       } else {
           return (
               <div>No Celebration Day is scheduled currently, Try again later</div>
           )
       }
    }

       componentDidMount(){
          this.props.getCelebrationDay();
        }

       

      render(){
        const { celebrationDay } = this.props;
    return (
            <Sports>
        
        <div className="sportsItem2"> 
        <div id="celebrationBackground">   
            <div  style={{backgroundColor:'gold'}} className="centredLatest">Celebration Day</div>
            
                {this.showMessages(celebrationDay)}
                 
            </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getCelebrationDay,
    deleteCelebrationDay
}

const mapStateToProps = (state) => ({
    celebrationDay:state.sports.celebrationDay,
    auth:state.auth.isAuthenticated
})
 
export default connect(mapStateToProps, actions)(CelebrationDay);