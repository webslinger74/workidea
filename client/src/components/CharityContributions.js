import React, { Component } from 'react';
import { connect } from 'react-redux';
import Sports from './Sports';
import { getCharity, deleteCharity } from '../actions/sportsActions';

class CharityContributions extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

         componentDidMount(){
             this.props.getCharity();
         }

         getGrandTotal = () => {
             var grandTotal = 0;
             this.props.charities.map( (charity) => {
                 grandTotal += parseInt(charity.amount)
             })
             return grandTotal;
         }

      render(){
            const { charities } = this.props;
    return (
            <Sports>
        <div className="charityBackground">    
        <div>
        <h1  style={{backgroundColor:'gold'}}>Money Raised this Year For Charity: £{(this.props.charities && this.props.charities) ? this.getGrandTotal() : 0.00}</h1>
           
            <p>Please Check out how much YOU have helped the S&Social raise through site events this year!</p>  

                        {charities && charities ? (
                            <div className="charityContainer">
                                {charities.map((charity) => {
                                  return  (
                                        <div key={charity._id} className="charityItems">
                                         Total Raised for: <span style={{fontSize:'40px', color:'green'}}>{charity.charity}</span> = <strong style={{fontSize:'40px'}}>£{charity.amount}</strong> 
                                         <div onClick={()=> this.props.deleteCharity({id:charity._id})}><span style={{fontSize:'12px'}}>DELETE</span></div>
                                        </div>
                                    )
                                }

                                )}

                            </div>
                        ):null}





        </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getCharity,
    deleteCharity
}

const mapStateToProps = (state) => ({
    charities:state.sports.charities
})
 
export default connect(mapStateToProps, actions)(CharityContributions);