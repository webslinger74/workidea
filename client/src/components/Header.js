import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';


export class Header extends Component {
    constructor(props) {
        super(props);
     
        this.state = {
            width:window.innerWidth,
            active:false,
            page: [
                {
                    name:'Home',
                    linkTo:'/',
                    public: true
                },
                {
                    name:'Messages',
                    linkTo:'/messageboard',
                    public: true
                },
                {
                    name:'Sports & Social',
                    linkTo:'/events',
                    public: true
                },
                {
                    name:'Well-Being',
                    linkTo:'/wellbeing/Events',
                    public: true
                },
                {
                    name:'Site Performance',
                    linkTo:'/sitePerformance',
                    public: true
                },
                {
                    name:'Manager',
                    linkTo:'/manager',
                    public: true
                },
                {
                    name:'Feedback',
                    linkTo:'/feedback',
                    public: true
                }
            ],
            user:[
                {
                    name: 'My Cart',
                    linkTo:'/user/cart',
                    public:false
                },
                {
                    name:'My Account',
                    linkTo:'/user/dashboard',
                    public:false
                },
                {
                    name:'Admin',
                    linkTo:'/login',
                    public:true
                },
                {
                    name:'Log Out',
                    linkTo:'/user/logout',
                    public:false
                }

            ]
        }

    }

    componentDidMount(){

        
        window.addEventListener('resize', ()=> {
            console.log("changed");
            this.setState({active:false});
            this.windowWidth();
        })
        window.addEventListener('load', ()=> {
            this.setState({active:false});
            this.windowWidth();
        })
        window.addEventListener('click', ()=> {
            this.windowWidth();
        })
       
       
        console.log(this.state.width);

    }

    componentDidUpdate(){
     
    }
    componentWillUnmount(){
       window.removeEventListener('resize', this.handleResize);
    }

    handleResize = () => this.forceUpdate();

    showLinks = (type) => {
        let list = [];
        if(!this.props.auth){
            console.log("not logged in")
            if(type === this.state.user){
                type.filter(link => {
               if(link.public === true){
                   list.push(link)
               }    
                })
                console.log(list, "this is the list of links after filter")
               const newlist =  list.map((item, i) =>  {
                  return this.defaultLinks(item, i)
              })
              console.log(newlist, "this is the post list")
              return newlist;



                 } else {
                    if(type === this.state.page){
                        const pagelist = type.map((item,i) => {
                            return this.defaultLinks(item,i)
                        })
                        console.log(pagelist, "this is the post list of public links")
                        return pagelist
                    }                 
                }           
        } else {



            if(type === this.state.user){
                type.filter(link => {
                    if(link.public === false){
                        list.push(link)
                    }
                })
                const newlist = list.map((item, i) => {
                    if(item.name !== "My Cart"){
                    return this.defaultLinks(item,i);
                    } else {
                        return this.cartLink(item,i);
                    }
                });
                console.log(newlist, "this is the post list")
                return newlist;
            }
            else {
                if(type === this.state.page){
                    const pagelist = type.map((item,i) => {
                        return this.defaultLinks(item,i)
                    })
                    console.log(pagelist, "this is the post list of public links")
                    return pagelist
                }  
            }   
        }
    }

    cartLink = (item,i) => {
        if(this.props.currentUser){
        const  currentUser  = this.props.currentUser;
        console.log(currentUser, "this is the user from in the component");
        
        return (
            <div className="cart_link" key={i}>
                <span>{currentUser.cart ? currentUser.cart.length : 0}</span>
                <Link to={item.linkTo}>
                    {item.name}
                </Link>
            </div>
        )
    }
}

    defaultLinks = (item, i) => (
        item.name === 'Log Out' ?
            <div className="log_out_link"
                key={i}
                onClick={() => this.props.logoutUser(this.props.history)}
                >

                 {item.name}

            </div>:
                 <Link key={i} to={item.linkTo}>{item.name}</Link>
        
                 )


    toggleClass = () => {
        const currentState = this.state.active;
        this.setState({
            active:!currentState
        })
    }
    
    windowWidth = () => {
        console.log(window.innerWidth,"the window width");
        if (window.innerWidth <= 850)
            return (
                <div>
            <div onClick={this.toggleClass}>
            Menu
            </div>
            {this.state.active ? <div className="bottom">{this.showLinks(this.state.page)}</div>
            :<div className="hiddenMenus"></div>}
            </div>
            )
            
        else 
            return (
                
                <div className='bottom'>
            
            
            {this.showLinks(this.state.page)}


            </div>
            )
        
           }
        
  
                    

  
  
    
 
    render() { 


        return (
            <header className="bck_b_light flowers">
            
            <div className="left">
            <div className="logo">
            CHORLTON
            </div>
            </div>
            <div className="right">
            <div className="top">

            {this.showLinks(this.state.user)}
            </div>
            
            {this.windowWidth()}
            

            </div>

            </header>
          );
    }
}
 

const mapStateToProps = (state) => ({
  //  auth: state.auth.isAuthenticated,
  //  currentUser: state.auth.FullUserRecord
})

const actions = {
    
}

export default connect(mapStateToProps, actions)(withRouter(Header));




