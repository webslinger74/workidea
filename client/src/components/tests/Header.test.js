import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';
import setUpTest from '../../setUpTest';



it('should render the Header component', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
})