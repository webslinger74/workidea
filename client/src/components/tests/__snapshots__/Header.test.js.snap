// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="left"
        >
          <div
            className="logo"
          >
            CHORLTON
          </div>
        </div>
        <div
          className="right"
        >
          <div
            className="top"
          >
            <Link
              replace={false}
              to="/login"
            >
              Admin
            </Link>
          </div>
          <div
            className="bottom"
          >
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              replace={false}
              to="/messageboard"
            >
              Messages
            </Link>
            <Link
              replace={false}
              to="/sports"
            >
              Sports & Social
            </Link>
            <Link
              replace={false}
              to="/achievements"
            >
              Achievements
            </Link>
            <Link
              replace={false}
              to="/sitePerformance"
            >
              Site Performance
            </Link>
            <Link
              replace={false}
              to="/manager"
            >
              Manager
            </Link>
            <Link
              replace={false}
              to="/feedback"
            >
              Feedback
            </Link>
          </div>
        </div>
      </div>,
      "className": "bck_b_light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="left"
          >
            <div
              className="logo"
            >
              CHORLTON
            </div>
          </div>,
          <div
            className="right"
          >
            <div
              className="top"
            >
              <Link
                replace={false}
                to="/login"
              >
                Admin
              </Link>
            </div>
            <div
              className="bottom"
            >
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
              <Link
                replace={false}
                to="/messageboard"
              >
                Messages
              </Link>
              <Link
                replace={false}
                to="/sports"
              >
                Sports & Social
              </Link>
              <Link
                replace={false}
                to="/achievements"
              >
                Achievements
              </Link>
              <Link
                replace={false}
                to="/sitePerformance"
              >
                Site Performance
              </Link>
              <Link
                replace={false}
                to="/manager"
              >
                Manager
              </Link>
              <Link
                replace={false}
                to="/feedback"
              >
                Feedback
              </Link>
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="logo"
            >
              CHORLTON
            </div>,
            "className": "left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CHORLTON",
              "className": "logo",
            },
            "ref": null,
            "rendered": "CHORLTON",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="top"
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Admin
                </Link>
              </div>,
              <div
                className="bottom"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
                <Link
                  replace={false}
                  to="/messageboard"
                >
                  Messages
                </Link>
                <Link
                  replace={false}
                  to="/sports"
                >
                  Sports & Social
                </Link>
                <Link
                  replace={false}
                  to="/achievements"
                >
                  Achievements
                </Link>
                <Link
                  replace={false}
                  to="/sitePerformance"
                >
                  Site Performance
                </Link>
                <Link
                  replace={false}
                  to="/manager"
                >
                  Manager
                </Link>
                <Link
                  replace={false}
                  to="/feedback"
                >
                  Feedback
                </Link>
              </div>,
            ],
            "className": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/login"
                  >
                    Admin
                  </Link>,
                ],
                "className": "top",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Admin",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Admin",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                  <Link
                    replace={false}
                    to="/messageboard"
                  >
                    Messages
                  </Link>,
                  <Link
                    replace={false}
                    to="/sports"
                  >
                    Sports & Social
                  </Link>,
                  <Link
                    replace={false}
                    to="/achievements"
                  >
                    Achievements
                  </Link>,
                  <Link
                    replace={false}
                    to="/sitePerformance"
                  >
                    Site Performance
                  </Link>,
                  <Link
                    replace={false}
                    to="/manager"
                  >
                    Manager
                  </Link>,
                  <Link
                    replace={false}
                    to="/feedback"
                  >
                    Feedback
                  </Link>,
                ],
                "className": "bottom",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Messages",
                    "replace": false,
                    "to": "/messageboard",
                  },
                  "ref": null,
                  "rendered": "Messages",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sports & Social",
                    "replace": false,
                    "to": "/sports",
                  },
                  "ref": null,
                  "rendered": "Sports & Social",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Achievements",
                    "replace": false,
                    "to": "/achievements",
                  },
                  "ref": null,
                  "rendered": "Achievements",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Site Performance",
                    "replace": false,
                    "to": "/sitePerformance",
                  },
                  "ref": null,
                  "rendered": "Site Performance",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Manager",
                    "replace": false,
                    "to": "/manager",
                  },
                  "ref": null,
                  "rendered": "Manager",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "class",
                  "props": Object {
                    "children": "Feedback",
                    "replace": false,
                    "to": "/feedback",
                  },
                  "ref": null,
                  "rendered": "Feedback",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="left"
          >
            <div
              className="logo"
            >
              CHORLTON
            </div>
          </div>
          <div
            className="right"
          >
            <div
              className="top"
            >
              <Link
                replace={false}
                to="/login"
              >
                Admin
              </Link>
            </div>
            <div
              className="bottom"
            >
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
              <Link
                replace={false}
                to="/messageboard"
              >
                Messages
              </Link>
              <Link
                replace={false}
                to="/sports"
              >
                Sports & Social
              </Link>
              <Link
                replace={false}
                to="/achievements"
              >
                Achievements
              </Link>
              <Link
                replace={false}
                to="/sitePerformance"
              >
                Site Performance
              </Link>
              <Link
                replace={false}
                to="/manager"
              >
                Manager
              </Link>
              <Link
                replace={false}
                to="/feedback"
              >
                Feedback
              </Link>
            </div>
          </div>
        </div>,
        "className": "bck_b_light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="left"
            >
              <div
                className="logo"
              >
                CHORLTON
              </div>
            </div>,
            <div
              className="right"
            >
              <div
                className="top"
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Admin
                </Link>
              </div>
              <div
                className="bottom"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
                <Link
                  replace={false}
                  to="/messageboard"
                >
                  Messages
                </Link>
                <Link
                  replace={false}
                  to="/sports"
                >
                  Sports & Social
                </Link>
                <Link
                  replace={false}
                  to="/achievements"
                >
                  Achievements
                </Link>
                <Link
                  replace={false}
                  to="/sitePerformance"
                >
                  Site Performance
                </Link>
                <Link
                  replace={false}
                  to="/manager"
                >
                  Manager
                </Link>
                <Link
                  replace={false}
                  to="/feedback"
                >
                  Feedback
                </Link>
              </div>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="logo"
              >
                CHORLTON
              </div>,
              "className": "left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CHORLTON",
                "className": "logo",
              },
              "ref": null,
              "rendered": "CHORLTON",
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="top"
                >
                  <Link
                    replace={false}
                    to="/login"
                  >
                    Admin
                  </Link>
                </div>,
                <div
                  className="bottom"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                  <Link
                    replace={false}
                    to="/messageboard"
                  >
                    Messages
                  </Link>
                  <Link
                    replace={false}
                    to="/sports"
                  >
                    Sports & Social
                  </Link>
                  <Link
                    replace={false}
                    to="/achievements"
                  >
                    Achievements
                  </Link>
                  <Link
                    replace={false}
                    to="/sitePerformance"
                  >
                    Site Performance
                  </Link>
                  <Link
                    replace={false}
                    to="/manager"
                  >
                    Manager
                  </Link>
                  <Link
                    replace={false}
                    to="/feedback"
                  >
                    Feedback
                  </Link>
                </div>,
              ],
              "className": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/login"
                    >
                      Admin
                    </Link>,
                  ],
                  "className": "top",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Admin",
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Admin",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>,
                    <Link
                      replace={false}
                      to="/messageboard"
                    >
                      Messages
                    </Link>,
                    <Link
                      replace={false}
                      to="/sports"
                    >
                      Sports & Social
                    </Link>,
                    <Link
                      replace={false}
                      to="/achievements"
                    >
                      Achievements
                    </Link>,
                    <Link
                      replace={false}
                      to="/sitePerformance"
                    >
                      Site Performance
                    </Link>,
                    <Link
                      replace={false}
                      to="/manager"
                    >
                      Manager
                    </Link>,
                    <Link
                      replace={false}
                      to="/feedback"
                    >
                      Feedback
                    </Link>,
                  ],
                  "className": "bottom",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Home",
                      "replace": false,
                      "to": "/",
                    },
                    "ref": null,
                    "rendered": "Home",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Messages",
                      "replace": false,
                      "to": "/messageboard",
                    },
                    "ref": null,
                    "rendered": "Messages",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sports & Social",
                      "replace": false,
                      "to": "/sports",
                    },
                    "ref": null,
                    "rendered": "Sports & Social",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "3",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Achievements",
                      "replace": false,
                      "to": "/achievements",
                    },
                    "ref": null,
                    "rendered": "Achievements",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "4",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Site Performance",
                      "replace": false,
                      "to": "/sitePerformance",
                    },
                    "ref": null,
                    "rendered": "Site Performance",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "5",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Manager",
                      "replace": false,
                      "to": "/manager",
                    },
                    "ref": null,
                    "rendered": "Manager",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "6",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Feedback",
                      "replace": false,
                      "to": "/feedback",
                    },
                    "ref": null,
                    "rendered": "Feedback",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
